sympy/solvers/solveset.py

```diff
@@ -539,23 +539,7 @@ def _solve_real_trig(f, symbol):
     """ Helper to solve trigonometric equations """
     f = trigsimp(f)
     f = f.rewrite(exp)
-    f = together(f)
-    g, h = fraction(f)
-    y = Dummy('y')
-    g, h = g.expand(), h.expand()
-    g, h = g.subs(exp(I*symbol), y), h.subs(exp(I*symbol), y)
-    if g.has(symbol) or h.has(symbol):
-        return ConditionSet(symbol, Eq(f, 0), S.Reals)
-
-    solns = solveset_complex(g, y) - solveset_complex(h, y)
-
-    if isinstance(solns, FiniteSet):
-        return Union(*[invert_complex(exp(I*symbol), s, symbol)[1]
-                       for s in solns])
-    elif solns is S.EmptySet:
-        return S.EmptySet
-    else:
-        return ConditionSet(symbol, Eq(f, 0), S.Reals)
```

Comment:
Not as I understand it. Though it is not enforced well, it is my understanding that the purpose of the `*_real` functions is to use the fact that the domain is real to select the correct/most appropriate solvers for the solving process. By allowing `f` to be solved in the complex domain you will allow this routine to send back complex solutions which is not desired. It's not a matter of "complication" it's a matter of what solution from the arbitrarily complex expression should be returned. 

However, seeing that there is not `_complex` trig routine and that the former function did the same, it seems that what you have done is no worse than what was present before.